### PR TITLE
Birth date pet screen

### PR DESCRIPTION
**Pull request by JorgeTranin**

### CODE REVIEW

**Comment by gusoliveira21** [comment]
@JorgeTranin Tente rodar sua aplicação em dispositivos diferentes no emulador:
Android 8.1 ("Oreo") | x86 API 27 apresentou erro.
FATAL EXCEPTION: main
Process: com.soujunior.petjournal, PID: 11495
java.lang.RuntimeException: Canvas: trying to draw too large(441000000bytes) bitmap.
	at android.view.DisplayListCanvas.throwIfCannotDraw(DisplayListCanvas.java:229)
	at android.view.RecordingCanvas.drawBitmap(RecordingCanvas.java:97)
	at androidx.compose.ui.graphics.AndroidCanvas.drawImageRect-HPBpro0(AndroidCanvas.android.kt:271)
	at androidx.compose.ui.graphics.drawscope.CanvasDrawScope.drawImage-AZ2fEMs(CanvasDrawScope.kt:263)
	at androidx.compose.ui.node.LayoutNodeDrawScope.drawImage-AZ2fEMs(Unknown Source:39)
	at androidx.compose.ui.graphics.drawscope.DrawScope.drawImage-AZ2fEMs$default(DrawScope.kt:510)
	at androidx.compose.ui.graphics.painter.BitmapPainter.onDraw(BitmapPainter.kt:93)
	at androidx.compose.ui.graphics.painter.Painter.draw-x_KDEd0(Painter.kt:212)
	at androidx.compose.ui.draw.PainterModifierNode.draw(PainterModifier.kt:347)
	at androidx.compose.ui.node.LayoutNodeDrawScope.draw-x_KDEd0$ui_release(LayoutNodeDrawScope.kt:92)
	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:370)
	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:359)
	at androidx.compose.ui.node.LayoutModifierNodeCoordinator.performDraw(LayoutModifierNodeCoordinator.kt:236)
	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:367)
	at androidx.compose.ui.node.NodeCoordinator.access$drawContainedDrawModifiers(NodeCoordinator.kt:58)
	at androidx.compose.ui.node.NodeCoordinator$invoke$1.invoke(NodeCoordinator.kt:396)
	at androidx.compose.ui.node.NodeCoordinator$invoke$1.invoke(NodeCoordinator.kt:395)
	at androidx.compose.runtime.snapshots.Snapshot$Companion.observe(Snapshot.kt:2200)
	at androidx.compose.runtime.snapshots.SnapshotStateObserver$observeReads$1$1.invoke(SnapshotStateObserver.kt:234)
	at androidx.compose.runtime.snapshots.SnapshotStateObserver$observeReads$1$1.invoke(SnapshotStateObserver.kt:230)
	at androidx.compose.runtime.SnapshotStateKt__DerivedStateKt.observeDerivedStateRecalculations(DerivedState.kt:341)
	at androidx.compose.runtime.SnapshotStateKt.observeDerivedStateRecalculations(Unknown Source:1)
	at androidx.compose.runtime.snapshots.SnapshotStateObserver.observeReads(SnapshotStateObserver.kt:230)
	at androidx.compose.ui.node.OwnerSnapshotObserver.observeReads$ui_release(OwnerSnapshotObserver.kt:120)
	at androidx.compose.ui.node.NodeCoordinator.invoke(NodeCoordinator.kt:395)
	at androidx.compose.ui.node.NodeCoordinator.invoke(NodeCoordinator.kt:58)
	at androidx.compose.ui.platform.RenderNodeApi23.record(RenderNodeApi23.android.kt:280)
	at androidx.compose.ui.platform.RenderNodeLayer.updateDisplayList(RenderNodeLayer.android.kt:301)
	at androidx.compose.ui.platform.RenderNodeLayer.drawLayer(RenderNodeLayer.android.kt:242)
	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:354)
	at androidx.compose.ui.node.LayoutModifierNodeCoordinator.performDraw(LayoutModifierNodeCoordinator.kt:236)
	at androidx.compose.ui.node.LayoutNodeDrawScope.drawContent(LayoutNodeDrawScope.kt:64)
	at androidx.compose.foundation.Background.draw(Background.kt:107)
	at androidx.compose.ui.node.BackwardsCompatNode.draw(BackwardsCompatNode.kt:361)
	at androidx.compose.ui.node.LayoutNodeDrawScope.draw-x_KDEd0$ui_release(LayoutNodeDrawScope.kt:92)
	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:370)
	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:359)
	at androidx.compose.ui.node.LayoutModifierNodeCoordinator.performDraw(LayoutModifierNodeCoordinator.kt:236)
	at androidx.compose.ui.node.NodeCoordinator.drawContainedDrawModifiers(NodeCoordinator.kt:367)
	at androidx.compose.ui.node.NodeCoordinator.draw(NodeCoordinator.kt:359)
	at androidx.compose.ui.node.LayoutNode.draw$ui_release(LayoutNode.kt:866)

File: petJournal/app/src/main/java/com/soujunior/petjournal/ui/navigation/MainActivity.kt [hunk 1]
```diff
@@ -14,7 +14,7 @@ class MainActivity : ComponentActivity() {
         super.onCreate(savedInstanceState)
         WindowCompat.setDecorFitsSystemWindows(window, false)
         setContent {
-            PresentationManager()
```
Comment (gusoliveira21) [inline petJournal/app/src/main/java/com/soujunior/petjournal/ui/navigation/MainActivity.kt]: ???

File: petJournal/app/src/main/java/com/soujunior/petjournal/ui/navigation/NavHostElements.kt [hunk 2]
```diff
@@ -69,9 +76,12 @@ fun NavHostMainContent() {
 @Composable
 fun NavTestScreen(){
```
Comment (gusoliveira21) [inline petJournal/app/src/main/java/com/soujunior/petjournal/ui/navigation/NavHostElements.kt]: Ainda é necessário manter o NavTestScreen() ? se não, pode remover

**Review by gusoliveira21 — CHANGES_REQUESTED** [review]
Remova métodos não utilizados como "success()".
Remova comentários (se o método ou classe não tiver em seu nome o objetivo claro do que ela faz, refatore para que ela faça isso).

File: petJournal/app/src/main/java/com/soujunior/petjournal/ui/appArea/pets/petNameAndGenderScreen/ViewModelNameGenderImpl.kt [hunk 3]
```diff
@@ -88,13 +92,37 @@ class ViewModelNameGenderImpl(
...
+                id = state.idPetInformation ?: 0L,
+                species = state.specie,
+                name = state.name,
+                gender = state.gender,
+                guardianId = 1
+            )
+
+            val result = updatePetInformationUseCase.execute(petInformation)
+            Log.i("tt", result.isFailure.toString())
+           // result.handleResult(::success, ::failed)
```
Comment (gusoliveira21) [inline petJournal/app/src/main/java/com/soujunior/petjournal/ui/appArea/pets/petNameAndGenderScreen/ViewModelNameGenderImpl.kt]: remover se não estiver sendo usado

File: petJournal/data/src/main/java/com/soujunior/data/repository/GuardianRepositoryImpl.kt [hunk 4]
```diff
@@ -64,5 +64,14 @@ class GuardianRepositoryImpl(
         }
     }
 
+    override suspend fun updatePetInformation(petInformationModel: PetInformationModel): DataResult<Unit> {
+        return try {
+            DataResult.Success(guardianLocalDataSourceImpl.updatePetInformation(petInformationModel).success.data)
+        }catch (e: Throwable){
+            Log.i("tt", e.message.toString())
```
Comment (gusoliveira21) [inline petJournal/data/src/main/java/com/soujunior/data/repository/GuardianRepositoryImpl.kt:71]: TAG

File: petJournal/app/src/main/java/com/soujunior/petjournal/ui/appArea/pets/petNameAndGenderScreen/ViewModelNameGenderImpl.kt [hunk 5]
```diff
@@ -29,7 +29,7 @@ class ViewModelNameGenderImpl(
     override val taskState: StateFlow<TaskState> = _taskState
 
     init {
-        getPetInformation(1)
+        //getPetInformation(state.idPetInformation!!)
```
Comment (gusoliveira21) [inline petJournal/app/src/main/java/com/soujunior/petjournal/ui/appArea/pets/petNameAndGenderScreen/ViewModelNameGenderImpl.kt:34]: ?

File: petJournal/app/src/main/java/com/soujunior/petjournal/ui/appArea/pets/petNameAndGenderScreen/ViewModelNameGenderImpl.kt [hunk 6]
```diff
@@ -29,7 +29,7 @@ class ViewModelNameGenderImpl(
     override val taskState: StateFlow<TaskState> = _taskState
 
     init {
-        getPetInformation(1)
```
Comment (gusoliveira21) [inline petJournal/app/src/main/java/com/soujunior/petjournal/ui/appArea/pets/petNameAndGenderScreen/ViewModelNameGenderImpl.kt]: ?

**Comment by gusoliveira21** [comment]
@JorgeTranin Refatora o código de modo a remover os comentários, códigos não utilizados e logs que não usem TAG, ou só adiciona TAG nelas. Tente remover também alguns métodos criados para testes como NavTestScreen() , generic(), ...

**Review by N0stalgiaUltra — CHANGES_REQUESTED** [review]
Screen.kt 
linhas 76, 83, 84, adicionar strings ao Strings Resources (strings.xml)

obs: não precisava remover o navhost de teste de telas. Eu deixava ele só pra n ter que alterar o metodo original, mas tb n tem problema ele ter sido removido.

**Comment by JorgeTranin** [comment]
Foi movida as Strings para o arquivo resource.

obs: Referente ao NavHost foi solicitado a remoção para a developer.

Valeu pelo feedback pessoal.

**Review by N0stalgiaUltra — APPROVED** [review]

**Pull request merged**